src/input.rs

```diff
@@ -94,15 +93,15 @@ pub trait Input: Sized {
     ) -> Self::SourceInfo;
     fn source_info_point<'i>(
         input: &Container<'i, Self::Container>,
-        index: Index<'i, Unknown>,
+        index: Index<'i, u32, Unknown>,
     ) -> Self::SourceInfoPoint;
 }
 
 impl<T> Input for &[T] {
     type Container = Self;
     type Slice = [T];
-    type SourceInfo = ops::Range<usize>;
-    type SourceInfoPoint = usize;
+    type SourceInfo = ops::Range<u32>;
```

Comment:
`indexing-str` is parameterized over the untrusted index type and defaults to `u32` rather than `usize` currently. Thus I've migrated to `u32` here for a smaller diff.